* use lrint() whenever the left-value is integer
this allows the compliation to use lrint() inlined math macros

without this patch, gcc 4.1.0 optimize rint() calls into lrint()
but, we dont have function of lrint(), just inlined math macro.

use lrintf() whenever the argument is float
for the same reason.

diff -r 51494c531769 libavcodec/snow.c
--- a/libavcodec/snow.c	Sun Apr 30 16:52:40 2006 +0200
+++ b/libavcodec/snow.c	Sun Apr 30 16:54:28 2006 +0200
@@ -3850,7 +3850,7 @@
 }
 
 static int qscale2qlog(int qscale){
-    return rint(QROOT*log(qscale / (float)FF_QP2LAMBDA)/log(2))
+    return lrint(QROOT*log(qscale / (float)FF_QP2LAMBDA)/log(2))
            + 61*QROOT/8; //<64 >60
 }
 
diff -r 51494c531769 libmpcodecs/vf_hue.c
--- a/libmpcodecs/vf_hue.c	Sun Apr 30 16:52:40 2006 +0200
+++ b/libmpcodecs/vf_hue.c	Sun Apr 30 16:54:28 2006 +0200
@@ -31,8 +31,8 @@ static void process_C(uint8_t *udst, uin
 		    int w, int h, float hue, float sat)
 {
 	int i;
-	const int s= rint(sin(hue) * (1<<16) * sat);
-	const int c= rint(cos(hue) * (1<<16) * sat);
+	const int s= lrint(sin(hue) * (1<<16) * sat);
+	const int c= lrint(cos(hue) * (1<<16) * sat);
 
 	while (h--) {
 		for (i = 0; i<w; i++)
@@ -110,10 +110,10 @@ static int control(struct vf_instance_s*
 	case VFCTRL_GET_EQUALIZER:
 		eq = data;
 		if (!strcmp(eq->item,"hue")) {
-			eq->value = rint(vf->priv->hue *100 / M_PI);
+			eq->value = lrint(vf->priv->hue *100 / M_PI);
 			return CONTROL_TRUE;
 		}else if (!strcmp(eq->item,"saturation")) {
-			eq->value = rint(vf->priv->saturation*100 - 100);
+			eq->value = lrintf(vf->priv->saturation*100 - 100);
 			return CONTROL_TRUE;
 		}
 		break;
diff -r 51494c531769 libvo/vo_dxr3.c
--- a/libvo/vo_dxr3.c	Sun Apr 30 16:52:40 2006 +0200
+++ b/libvo/vo_dxr3.c	Sun Apr 30 16:54:28 2006 +0200
@@ -1307,7 +1307,7 @@ static int col_interp(float x, struct co
 	y = 255;
     if(y < 0)
 	y = 0;
-    return rint(y);
+    return lrintf(y);
 }
 
 static int overlay_set_keycolor(overlay_t *o, int color) {
diff -r 51494c531769 libavcodec/mpegaudiodec.c
--- a/libavcodec/mpegaudiodec.c	2006-12-10 04:02:54.000000000 +0100
+++ b/libavcodec/mpegaudiodec.c	2006-12-10 04:03:17.000000000 +0100
@@ -402,9 +402,9 @@
         for(i=0; i<512*16; i++){
             int exponent= (i>>4);
             double f= pow(i&15, 4.0 / 3.0) * pow(2, (exponent-400)*0.25 + FRAC_BITS + 5);
-            expval_table[exponent][i&15]= llrint(f);
+            expval_table[exponent][i&15]= lrint(f);
             if((i&15)==1)
-                exp_table[exponent]= llrint(f);
+                exp_table[exponent]= lrint(f);
         }
 
         for(i=0;i<7;i++) {
